### PR TITLE
Enforce strict equality operator

### LINKED ISSUE #1890: [Tech Debt]: Consider using ESLint to enforce strict equality operator
### Type

General Tech Debt (eg. Improve the way something exists today)

### Source

main

### Description

Currently the linter doesn't enforce use of the strict equality operator. We should consider setting the `eqeqeq` lint rule to enforce this.

### Why?

It is considered good practice to use the type-safe equality operators === and !== instead of their regular counterparts == and !=.

The reason for this is that == and != do type coercion which follows the rather obscure [Abstract Equality Comparison Algorithm](https://www.ecma-international.org/ecma-262/5.1/#sec-11.9.3). For instance, the following statements are all considered true:

[] == false
[] == ![]
3 == "03"
If one of those occurs in an innocent-looking statement such as a == b the actual problem is very difficult to spot.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by manaswinidas**
Fixes #1890 

## Description

Enforce strict equality operator by including appropriate rules in `.eslintrc` according to [`eqeqeq` documentation](https://eslint.org/docs/latest/rules/eqeqeq) and made changes to accommodate the rule.

## How Has This Been Tested?

Run `npm run test`. Any outstanding `!=` or `==` will be detected.

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: backend/.eslintrc [hunk 1]
```diff
@@ -32,13 +30,16 @@
...
   },
-  "settings": {
-  },
+  "settings": {},
   "rules": {
-    "no-restricted-properties": [ "error", {
-      "property": "toString",
-      "message": "e.toString() should be fastify.log.error(e, 'your string'). Other use-cases should avoid obj.toString() on principle. Craft the string you want instead."
-    }],
+    "eqeqeq": ["error", "always"],
```
Comment (manaswinidas) [inline backend/.eslintrc:35]: These are some other options we can use: https://eslint.org/docs/latest/rules/eqeqeq#options

Comment (christianvogt) [reply backend/.eslintrc:35]: @manaswinidas please include option `{ null: 'ignore' }`. We shouldn't have to distinguish between `undefined` and `null`.

File: frontend/src/pages/BYONImages/UpdateImageModal.tsx [hunk 2]
```diff
@@ -39,13 +39,13 @@ export const UpdateImageModal: React.FC<UpdateImageModalProps> = ({
 }) => {
   const [name, setName] = React.useState<string>(image.name);
   const [description, setDescription] = React.useState<string>(
-    image.description != undefined ? image.description : '',
+    image.description !== undefined ? image.description : '',
```
Comment (christianvogt) [inline frontend/src/pages/BYONImages/UpdateImageModal.tsx:42]: Note sure if `image.description` can ever equal `null`, but this change has potentially different outcomes. I'd simplify this to the following (applies to other changes as well):
```suggestion
    image.description ?? '',
```

**Comment by shalberd** [comment]
thank you @manaswinidas

**Pull request merged**